Update Documentation for Internal Metric Naming Conventions (#5178)
open-telemetry/opentelemetry.io: content/en/docs/collector/internal-telemetry.md

emitted. For the complete list of metrics, with a breakdown by level, see
[Lists of internal metrics](#lists-of-internal-metrics).

> > **Note:** As of Collector version v0.106.1, internal metric names are handled differently based on their source:
> > - Metrics generated from Collector components are prefixed with `otelcol_`.
> > - Metrics generated from instrumentation libraries do not use the `otelcol_` prefix.
> > 
> > For Collector versions prior to v0.106.1, all internal metrics, regardless of their origin, were prefixed with `otelcol_`. This includes metrics from both Collector components and instrumentation libraries.

Reviewer: ```suggestion
{{% alert title="Note" color="info" %}}
As of Collector v0.106.1, internal metric names are handled differently based on their source:
- Metrics generated from Collector components are prefixed with `otelcol_`.
- Metrics generated from instrumentation libraries do not use the `otelcol_` prefix.

For Collector versions prior to v0.106.1, all internal metrics, regardless of their origin, are prefixed with `otelcol_`. This includes metrics from both Collector components and instrumentation libraries.
{{% /alert %}}
```

  telemetry:
    metrics:
      level: detailed

> 

Reviewer: We'll need these back ticks to close the yaml code block. Could you add them again?

`normal`, and `detailed`. Each metric is identified by name and description and
categorized by instrumentation type.

{{% alert title="Note" color="info" %}} As of Collector v0.106.1, internal
metric names are handled differently based on their source:

- Metrics generated from Collector components are prefixed with `otelcol_`.
- Metrics generated from instrumentation libraries do not use the `otelcol_`
  prefix.

> For Collector versions prior to v0.106.1, all internal metrics, regardless of

Reviewer 2: ```suggestion
For Collector versions prior to v0.106.1, all internal metrics emitted via the Prometheus exporter, regardless of
```

> For Collector versions prior to v0.106.1, all internal metrics emitted via the Prometheus exporter, regardless of

Reviewer: ```suggestion
For Collector versions prior to v0.106.1, all internal metrics emitted using the Prometheus exporter, regardless of
```
